lib/sanity/attributable.rb

```diff
@@ -22,33 +29,46 @@ def attributes
         @attributes ||= []
       end
 
-      def default_attributes
-        @defaults ||= {}
+      def aliased_attributes
+        @aliased_attributes ||= {}
       end
 
       private
 
-      def attribute(name, default: nil)
-        attributes << name
-        default_attributes.merge!("#{name}": default)
+      def attribute(name, default: nil, as: nil, type: nil)
+        attributes.push(name)
+
+        self.define_method("#{name}=") do |value|
+          @attributes[name] =
+            case type
+            when :boolean
+              value.to_s == 'true'
+            when :float
+              value.to_f
+            when :integer
+              value.to_i
+            when :time
+              Time.parse(value.to_s)
+            else
+              value
+            end
+        end
+        self.define_method("#{name}") { @attributes[name] }
+
+        if as
+          aliased_attributes[as] = name
+          self.alias_method("#{as}=", "#{name}=")
+          self.alias_method(as, name)
+        end
       end
+
     end
 
     attr_reader :attributes
 
-    def initialize(**args)
-      self.class.default_attributes.merge(args).then do |attrs|
-        attrs.each do |key, val|
-          define_singleton_method("#{key}=") do |val|
-            args[key] = val
-            attributes[key] = val
-          end
-
-          define_singleton_method(key) { args[key] }
-        end
-
-        instance_variable_set(:@attributes, attrs)
-      end
```

Comment:
By removing this - you've broken the ability to initialize a new instance of an object with key/value pairs that aren't defined as an `attribute`.

This is a breaking change and I'm on the fence about whether it's a good change for this gem or not. What was your thinking behind this?

Reply:
I see what you mean. I think my other comment explains my reasoning. If I was making this gem from the start, my goal would be to make it feel as much like ActiveModel as possible since that makes it easier to use. That's just my 2 cents.